pandas/tseries/frequencies.py

```diff
@@ -298,6 +304,28 @@ def to_offset(freqstr):
             name, stride = stride, name
         name, _ = _base_and_stride(name)
         delta = get_offset(name) * stride
+
+    elif isinstance(freqstr, tslib.Timedelta):
+        name_to_offset_map = {'days': Day(1),
+                              'hours': Hour(1),
+                              'minutes': Minute(1),
+                              'seconds': Second(1),
+                              'milliseconds': Milli(1),
+                              'microseconds': Micro(1),
+                              'nanoseconds': Nano(1)}
+        try:
+            for name in freqstr.components._fields:
+                offset = name_to_offset_map[name]
+                stride = getattr(freqstr.components, name)
+                if stride is not None and int(stride) != 0:
```

Comment:
yes this cannot be ``None``. Its always an int (but of course you only want to consider non-zero ones as you are doing).